### PR TITLE
PHOENIX-6864 - create view throwing NPE when referring back to itself

### PR DESCRIPTION
**Pull request by Divneet18**
Changes made in this PR :- Throwing a well defined error when trying to create a view with the same name as parent table/view

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/compile/CreateTableCompiler.java [hunk 1]
```diff
@@ -55,15 +55,8 @@
 import org.apache.phoenix.parse.SelectStatement;
 import org.apache.phoenix.parse.TableName;
 import org.apache.phoenix.query.QueryConstants;
-import org.apache.phoenix.schema.ColumnRef;
-import org.apache.phoenix.schema.MetaDataClient;
-import org.apache.phoenix.schema.PDatum;
-import org.apache.phoenix.schema.PName;
-import org.apache.phoenix.schema.PTable;
+import org.apache.phoenix.schema.*;
```
Comment (lokiore) [inline phoenix-core/src/main/java/org/apache/phoenix/compile/CreateTableCompiler.java:58]: Can we revert changes in this file as they are not needed!?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIT.java [hunk 2]
```diff
@@ -56,9 +56,7 @@
 import org.apache.phoenix.jdbc.PhoenixStatement;
 import org.apache.phoenix.query.KeyRange;
 import org.apache.phoenix.query.QueryServices;
-import org.apache.phoenix.schema.PTable;
-import org.apache.phoenix.schema.PTableImpl;
-import org.apache.phoenix.schema.ReadOnlyTableException;
+import org.apache.phoenix.schema.*;
```
Comment (lokiore) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIT.java:59]: Please remove * import!

**Review by lokiore — APPROVED** [review]
Couple of nits otherwise looks good!

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIT.java [hunk 3]
```diff
@@ -896,6 +899,66 @@ public void testCompositeDescPK() throws Exception {
...
+            String fullViewName = generateUniqueName();
+            String ddl = "CREATE VIEW " + fullViewName +
+                    " AS SELECT * FROM " + fullViewName +
+                    " WHERE " + fullViewName + " = 1";
+            stmt.execute(ddl);
+            fail("Should have thrown an exception");
+        } catch (ColumnNotFoundException columnException) {
+            assertEquals("Undefined column",
+                    SQLExceptionCode.COLUMN_NOT_FOUND
+                    .getErrorCode(), columnException.getErrorCode());
```
Comment (virajjasani) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIT.java:917]: let's also assert the value of `ColumnNotFoundException#getColumnName`?

**Review by virajjasani — APPROVED** [review]
left a nit, +1 otherwise

**Review by virajjasani — APPROVED** [review]
+1

**Pull request merged**